jvm/ml4ir-inference/pom.xml

```diff
@@ -0,0 +1,183 @@
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>ml4ir</groupId>
+    <artifactId>ml4ir-parent</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
+  </parent>
+  <artifactId>ml4ir-inference</artifactId>
+  <packaging>jar</packaging>
+
+  <build>
+    <sourceDirectory>src/main/scala</sourceDirectory>
+    <testSourceDirectory>src/test/scala</testSourceDirectory>
+
+    <plugins>
+      <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-site-plugin</artifactId>
+          <version>3.7.1</version>
+      </plugin>
+      <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-project-info-reports-plugin</artifactId>
+          <version>3.0.0</version>
+      </plugin>
+      <!-- mixed scala/java compile -->
+      <plugin>
+	<groupId>net.alchim31.maven</groupId>
+	<artifactId>scala-maven-plugin</artifactId>
+        <version>3.3.1</version>
+	<executions>
+	  <execution>
+	    <id>compile</id>
+	    <goals>
+	      <goal>compile</goal>
+	    </goals>
+	    <phase>compile</phase>
+	  </execution>
+	  <execution>
+	    <id>test-compile</id>
+	    <goals>
+	      <goal>testCompile</goal>
+	    </goals>
+	    <phase>test-compile</phase>
+	  </execution>
+	  <execution>
+	    <phase>process-resources</phase>
+	    <goals>
+	      <goal>compile</goal>
+	    </goals>
+	  </execution>
+	</executions>
+      </plugin>
+      <plugin>
+	<artifactId>maven-compiler-plugin</artifactId>
+	<configuration>
+	  <source>1.7</source>
+	  <target>1.7</target>
+	</configuration>
+      </plugin>
+		<!-- disable surefire - - >
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-surefire-plugin</artifactId>
+          <version>2.7</version>
+          <configuration>
+            <skipTests>true</skipTests>
+          </configuration>
+        </plugin>
+        <! - - enable scalatest
+        <plugin>
+          <groupId>org.scalatest</groupId>
+          <artifactId>scalatest-maven-plugin</artifactId>
+          <version>1.0</version>
+          <configuration>
+            <reportsDirectory>${project.build.directory}/surefire-reports</reportsDirectory>
+            <junitxml>.</junitxml>
+            <filereports>WDF TestSuite.txt</filereports>
+          </configuration>
+          <executions>
+            <execution>
+              <id>test</id>
+              <goals>
+                <goal>test</goal>
+              </goals>
+            </execution>
+          </executions>
+        </plugin> -->
+              <!-- for fatjar -->
+      <plugin>
+	<groupId>org.apache.maven.plugins</groupId>
+	<artifactId>maven-assembly-plugin</artifactId>
+	<version>3.1.0</version>
+	<configuration>
+	  <descriptorRefs>
+	    <descriptorRef>jar-with-dependencies</descriptorRef>
+	  </descriptorRefs>
+	  <finalName>uber-ml4ir-parent-0.0.1-SNAPSHOT</finalName>
```

Comment:
I prefer not to use finalName in plugins that produces others maven artifact: you lose the auto versioning of names when you do that. Furthermore, using classifiers it more the maven way